### ISSUE TITLE
Add license files

### ISSUE THREAD

**Issue by MarkMcCaskey** [WebAssembly/WASI#182]
Hello! I was looking over the repo and realized that there aren't any license files:

~The `witx` tool has a license listed in the `cargo.toml`, but we should include a license file as well.~
I'd like to add a license for the entire repo if possible, but especially for the docs and witx data files themselves.

I'm not aware of if there's any formal process required for this or if everything in the WebAssembly org is assumed to be under a certain license.

Apache 2 seems reasonable to me though -- it's what witx and some other projects in the WebAssembly org are using.  If that sounds reasonable to you all, I can make a pull request to add it!

edit:

looks like there's been a need to use a separate documentation license in the past https://github.com/WebAssembly/spec/pull/496.  That raises the question of whether witx is documentation or source code...

edit2: not sure how I missed the license file on witx -- thanks for the correction!

**Comment by programmerjake** [comment]
there are some license files: https://github.com/WebAssembly/WASI/blob/master/tools/witx/LICENSE
there may be more that I missed.

**Comment by sunfishcode** [comment]
Yes, the `witx` library already has a license. For the `witx` files themselves, I'm seeking input on the proper way to proceed here.

**Comment by decathorpe** [comment]
Has anybody looked at this lately?

I'm working on packaging wasmtime for Fedora Linux, which includes a copy of the `witx` files from this repository, but inclusion of the packages is blocked for now, because it is not specified how the files in this project are licensed.

**Comment by tschneidereit** [comment]
I'm pretty sure `witx` files aren't copyrightable in any legal domain, because of how rigidly structured they are, and in the US specifically they seem to [quite clearly fall under fair use](https://arstechnica.com/tech-policy/2021/04/supreme-court-sides-with-google-in-api-copyright-battle-with-oracle/), but to eliminate all uncertainty I'd propose we explicitly license them under CC0 / public domain.

**Comment by font** [comment]
> I'm pretty sure `witx` files aren't copyrightable in any legal domain, because of how rigidly structured they are, and in the US specifically they seem to [quite clearly fall under fair use](https://arstechnica.com/tech-policy/2021/04/supreme-court-sides-with-google-in-api-copyright-battle-with-oracle/), but to eliminate all uncertainty I'd propose we explicitly license them under CC0 / public domain.

Could we consider a different license such as [MIT-0](https://opensource.org/licenses/MIT-0)? There's been a change in classification of CC0 to only be used for content and not code. The use of CC0 for code will prevent us from being able to officially package in Fedora. See https://lists.fedoraproject.org/archives/list/legal@lists.fedoraproject.org/thread/RRYM3CLYJYW64VSQIXY6IF3TCDZGS6LM/ for details.

**Comment by decathorpe** [comment]
It looks like license / copyright text was added here:
https://github.com/WebAssembly/WASI/commit/a334bc8

This references the [W3C Community Contributor License Agreement (CLA) Deed](https://www.w3.org/community/about/process/cla-deed/), which doesn't look like a standard open source license.

Does anybody know if this has been submitted to SPDX, or to the OSI or FSF for classification as a free / open source software license?

**Comment by ieure** [comment]
Same question here.  I'm trying to get wasi-libc packaged for [GNU Guix](https://guix.gnu.org), but the unusual licensing of this repo is a blocker.

I found [some discussion at the ASF](https://issues.apache.org/jira/browse/LEGAL-241?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=15257277), which concluded that it was a permissive license acceptable for their use.  However:

- The CLA is clearly *mostly* for those participating in W3C processes, not end users.
- The clause about withdrawing contributions up to 45 days after they've been published is incompatible with the GPL, which states "All rights granted under this License are granted for the term of copyright on the Program, and are irrevocable..."
- It's not in SPDX.
- Multiple independent groups have been confused by the licensing here.

If there's consensus within the WASI project that the .witx / .wit files are non-copyrightable, a simple statement that these files are in the public domain in a LICENSE file would go a long way to clarifying the situation.
